services/src/pro/users/postgres_userdb.rs

```diff
@@ -425,6 +435,17 @@ where
             .await
             .map_err(|_error| error::Error::InvalidSession)?;
 
+        let valid_session: bool = row.get(7);
+
+        let valid_until = if valid_session {
+            row.get(4)
+        } else {
+            self.refresh_tokens(session, &tx)
```

Comment:
don't you have to check whether the refresh token is still valid or where does this happen?

Reply:
The Identity Provider does not send a expiration date with the refresh token. The refresh token might be invalid for a variety of reasons (expired, revoked, ...). The only way to check is to send it to the Identity Provider.